docs/src/algorithms/clipping.md

```diff
@@ -32,4 +32,31 @@ viz(poly)
 viz!(other, color = :black, alpha = 0.2)
 viz!(boundary(clipped), color = :red, segmentsize = 3)
 Mke.current_figure()
-```
\ No newline at end of file
+```
+
+## WeilerAtherton
+
+```@docs
+WeilerAtherton
+```
+
+```@example clipping
+# polygon to clip
+outer = Ring((8, 0), (4, 8), (2, 8), (-2, 0), (0, 0), (1, 2), (5, 2), (6, 0))
+inner = Ring((4, 4), (2, 4), (3, 6))
+poly = PolyArea([outer, inner])
+
+# clipping geometry can be a polygon with holes
+other = poly |> Rotate(pi) |> Translate(8.0, 8.0)
+
+# clipped polygon
+clipped = clip(poly, other, WeilerAtherton())
+
+viz(poly)
+viz!(other, color = :black, alpha = 0.2)
+viz!(boundary(clipped), color = :red, segmentsize = 3)
+Mke.current_figure()
+```
+
+Algorithm is by default limited with a maximum number of vertices of 50 for each ring of the resulting polygon.
+The limit can be adjusted using a ScopedValue WEILER_ATHERTON_MAX_VERTICES_NUM.
```

Comment:
Can you please comment on why we need this limit?

Reply:
This was something needed during the development, when traversing the data structure to collect new output rings the algorithm could end up in infinite cycles. This should not happen anymore. I did not test on broken and self intersecting polygons, but it should not happen there either. I don't know why I added a documentation for it instead of removing it in the first place. Now it is removed.